cli/cmd/sources_cluster.go

```diff
@@ -0,0 +1,296 @@
+	_, excluded := excludedApps[app.GetName()]
+	if excluded {
+		fmt.Printf("    - Skipping %s due to exclusion file\n", kind)
+		return nil
+	}
+	if dryRun {
+		fmt.Printf("    - Dry-Run mode ENABLED - No changes will be made\n")
+		return nil
+	}
+	err := orchestrator.Apply(ctx, app)
+	return err
+}
+
+func instrumentNamespace(ctx context.Context, client *kube.Client, ns string, excludedApps map[string]struct{}, orchestrator *lifecycle.Orchestrator, dryRun bool) error {
+	deps, err := client.AppsV1().Deployments(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("  - \033[31mERROR\033[0m Cannot list deployments: %s\n", err)
+		return nil
+	}
+	for _, dep := range deps.Items {
+		err = instrumentApp(ctx, &dep, excludedApps, orchestrator, dryRun, "Deployment")
+		if errors.Is(err, context.Canceled) {
+			return err
+		}
+	}
+
+	// StatefulSets
+	statefulsets, err := client.AppsV1().StatefulSets(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("  - \033[31mERROR\033[0m Cannot list statefulsets: %s\n", err)
+		return nil
+	}
+	for _, sts := range statefulsets.Items {
+		err = instrumentApp(ctx, &sts, excludedApps, orchestrator, dryRun, "StatefulSet")
+		if errors.Is(err, context.Canceled) {
+			return err
+		}
+	}
+
+	// DaemonSets
+	daemonsets, err := client.AppsV1().DaemonSets(ns).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("  - \033[31mERROR\033[0m Cannot list daemonsets: %s\n", err)
+		return nil
+	}
+	for _, ds := range daemonsets.Items {
+		err = instrumentApp(ctx, &ds, excludedApps, orchestrator, dryRun, "DaemonSet")
+		if errors.Is(err, context.Canceled) {
+			return err
+		}
+	}
+
+	// CronJobs - handle both v1 and v1beta1
+	ver := cmdcontext.K8SVersionFromContext(ctx)
+	if ver.LessThan(version.MustParseSemantic("1.21.0")) {
+		// Use v1beta1 for Kubernetes < 1.21
+		cronjobs, err := client.BatchV1beta1().CronJobs(ns).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			fmt.Printf("  - \033[31mERROR\033[0m Cannot list cronjobs (v1beta1): %s\n", err)
+			return nil
+		}
+		for _, cj := range cronjobs.Items {
+			err = instrumentApp(ctx, &cj, excludedApps, orchestrator, dryRun, "CronJob (v1beta1)")
+			if errors.Is(err, context.Canceled) {
+				return err
+			}
+		}
+	} else {
+		// Use v1 for Kubernetes >= 1.21
+		cronjobs, err := client.BatchV1().CronJobs(ns).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			fmt.Printf("  - \033[31mERROR\033[0m Cannot list cronjobs (v1): %s\n", err)
+			return nil
+		}
+		for _, cj := range cronjobs.Items {
+			err = instrumentApp(ctx, &cj, excludedApps, orchestrator, dryRun, "CronJob (v1)")
+			if errors.Is(err, context.Canceled) {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func sliceToMap(slice []string) map[string]struct{} {
+	m := make(map[string]struct{})
+	for _, s := range slice {
+		m[s] = struct{}{}
+	}
+	return m
+}
+
+func readAppListFromFile(filename string) (map[string]struct{}, error) {
```

Comment:
since it is used both for apps and ns - maybe worth to rename the function

```suggestion
func readListFromFile(filename string) (map[string]struct{}, error) {
```